tests/test_filters.py

```diff
@@ -0,0 +1,346 @@
+        raster_array = get_array_and_mask(self.landsat_dem)[0]
+        raster_filtered = filter_func(raster_array)
+
+        if name in ("median", "mean"):
+            assert np.min(raster_array) <= np.min(raster_filtered)
+            assert np.max(raster_array) >= np.max(raster_filtered)
+        elif name == "min":
+            assert np.min(raster_array) == np.min(raster_filtered)
+            assert np.max(raster_array) >= np.max(raster_filtered)
+        elif name == "max":
+            assert np.min(raster_array) <= np.min(raster_filtered)
+            assert np.max(raster_array) == np.max(raster_filtered)
+
+        assert raster_array.shape == raster_filtered.shape
+
+        # Test with NaNs
+        nan_count = 1000
+        rng = np.random.default_rng(42)
+        cols = rng.integers(0, high=self.landsat_dem.width - 1, size=nan_count)
+        rows = rng.integers(0, high=self.landsat_dem.height - 1, size=nan_count)
+        raster_with_nans = np.copy(raster_array).squeeze()
+        raster_with_nans[rows, cols] = np.nan
+
+        raster_with_nans_filtered = filter_func(raster_with_nans)
+        if name in ("median", "mean"):
+            assert np.nanmin(raster_with_nans) <= np.nanmin(raster_with_nans_filtered)
+            assert np.min(raster_filtered) == np.nanmin(raster_with_nans_filtered)
+        elif name == "min":
+            assert np.nanmin(raster_with_nans) == np.nanmin(raster_with_nans_filtered)
+            assert np.min(raster_filtered) == np.nanmin(raster_with_nans_filtered)
+            assert np.nanmax(raster_with_nans) >= np.nanmax(raster_with_nans_filtered)
+        elif name == "max":
+            assert np.nanmin(raster_with_nans) <= np.nanmin(raster_with_nans_filtered)
+            assert np.nanmax(raster_with_nans) == np.nanmax(raster_with_nans_filtered)
+            assert np.max(raster_filtered) == np.nanmax(raster_with_nans_filtered)
+
+        if name != "mean":
+            array_3d = np.vstack((raster_array[np.newaxis, :], raster_array[np.newaxis, :]))
+            raster_filtered = filter_func(array_3d)
+            assert array_3d.shape == raster_filtered.shape
+            data = raster_array[:, 0]
+            pytest.raises(ValueError, filter_func, data)
+
+    def test_median_filter_nan_consistency(self) -> None:
+        """Test that different median filter engines return consistent results with NaNs."""
+        arr = np.array([[1, 2, np.nan], [4, np.nan, 6], [7, 8, 9]], dtype=np.float32)
+        filtered_scipy = gu.filters.median_filter(arr, window_size=3, engine="scipy")
+        filtered_numba = gu.filters.median_filter(arr, window_size=3, engine="numba")
+
+        assert filtered_scipy.shape == arr.shape
+        assert filtered_numba.shape == arr.shape
+        mask = np.isfinite(filtered_scipy) & np.isfinite(filtered_numba)
+        np.testing.assert_allclose(filtered_scipy[mask], filtered_numba[mask], rtol=1e-5)
```

Comment:
We shouldn't test on the mask but everywhere with `equal_nan=True` here, otherwise they could have different NaNs:
```python
assert np.allclose(filtered_scipy, filtered_numba, equal_nan=True)
```